src/webui/api/apicontroller.h

```diff
@@ -62,10 +63,10 @@ class APIController : public ApplicationComponent<QObject>
     const DataMap &data() const;
     void requireParams(const QVector<QString> &requiredParams) const;
 
-    void setResult(const QString &result);
-    void setResult(const QJsonArray &result);
-    void setResult(const QJsonObject &result);
-    void setResult(const QByteArray &result, const QString &mimeType = {}, const QString &filename = {});
+    void setResult(const QString &result, const int statusCode = 200);
```

Comment:
Will make this change. This seems like an arbitrary C++ thing, and at this point I'm pretty junior with the language. Is there somewhere I can read more on this pattern?

Reply:
The function declaration is only relevant to the caller. And the constness of the parameter (itself, not referred or pointed value) makes sense only inside the body of the function. For the caller, it is at least meaningless, but it can also confuse the reader under certain conditions. It is not prohibited by the compiler, but we prohibit it in our project coding style (many code analyzers also produce warnings about it).

Reply:
Thank you!